### PR TITLE
Readme overhaul

### LINKED ISSUE #738: Update readme
I am proposing an enhancement to our current README file. As the first point of interaction for users and potential contributors, a well-crafted README is important. It not only provides information about the project but also sets the tone for our community's culture and standards.

- For many, the README is the first introduction to our project. A clear README ensures a positive first impression and clear guidance on the information the user needs
- Ease of Use: A comprehensive README with proper setup, installation, and usage instructions makes it easier for new users to get started, reducing entry barriers.
- Clear structure It serves as a centralized source of information, communicating the project's purpose, scope, and how to contribute, thus fostering a more collaborative environment.

Proposed Enhancements:

- Structured Sections: Implement a structured layout with sections like Introduction, Installation, Usage, Features, Contributing, etc., for better readability and navigation.
- Detailed Setup Instructions: Provide clear and concise instructions suitable for various user levels.
- Contributing Guidelines: Clearly outline how potential contributors can get involved, report issues, and submit proposals.
- Project Status and Roadmap: Communicate the current state of the project future plans and link to our roadmap
- Acknowledgments and Licensing: Recognize contributors by featuring them on the readme and clearly state the project's licensing information.
- Add visuals, and link to important support pages (website, discord, etc)

### PR DESCRIPTION
**Pull request by pavlenex**
Closes #738 

Besides  of looking at diff, I think this readme can be [tested here on my fork a](https://github.com/pavlenex/stratum/tree/readme-update?tab=readme-ov-file#------stratum-v2-reference-implementation-sri--)s well, to get a better feeling on the visual changes.

I've removed a lot of stuff and tried to keep it consistent, if I removed something important, let me know. I believe a lot of what was removed was outdated or too specific and instead better suited for dev docs.

One thing I failed to figure out how to do properly is "Supporters" logos, I hope someone can help me with aligning them properly and scaling. 

Looking forward to feedback!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,545 +1,132 @@
...
+  <a href="#introduction">Introduction</a> •
+  <a href="#getting-started">Getting Started</a> •
+  <a href="#use-cases">Use Cases</a> •
+  <a href="#roadmap">Roadmap</a> •
+  <a href="#contribute">Contribute</a> •
+  <a href="#support">Support</a> •
+  <a href="#donate">Donate</a> •
+  <a href="#supporters">Supporters</a> •
+  <a href="#license">License</a> 
+  <a href="msrv">MSRV</a>
```
Comment (GitGab19) [inline README.md:33]: I think here one '#' is missing (?)
The link is not working on your fork

File: README.md [hunk 2]
```diff
@@ -1,545 +1,132 @@
...
+</p>
 
-The goals of this project are to provide:
+## 👋Introduction
```
Comment (GitGab19) [inline README.md:36]: I would introduce a space between emoji and text

File: README.md [hunk 3]
```diff
@@ -1,545 +1,132 @@
...
-- `binary_sv2`
-
-### 3.09 protocols/v2/sv2-ffi
-
-Exports a C static library with the minimum subset of the `protocols/v2` libraries required to
-build a Template Provider. Every dependency is compiled WITHOUT the `with_noise` and `with_serde`
-flags.
-
-**Internal dependencies**:
+### 👷Miners
```
Comment (GitGab19) [inline README.md:55]: I would add a space in between also here

File: README.md [hunk 4]
```diff
@@ -1,545 +1,132 @@
...
 
-- `codec_sv2`
-- `const_sv2`
-- `binary_sv2` (maybe in the future will use the one already imported by `codec_sv2`)
-- `subprotocols/common_messages_sv2`
-- `subprotocols/template_distribution_sv2`
+- Sv1 Miners can use the translator proxy (`roles/translator`) to connect with a Sv2-compatible pool.
+- Sv1 mining farms mining to a Sv2-compatible pool gain some of the security and efficiency improvements Sv2 offers over Stratum V1 (Sv1). The Sv1<->Sv2 translator proxy does not support  _all_ the features of Sv2, but works as a temporary measure before upgrading completely to Sv2-compatible firmware. (The Sv1<->Sv2 translation proxy implementation is a work in progress.)
```
Comment (GitGab19) [inline README.md:58]: Uppercase for SV1 and SV2 (?)

File: README.md [hunk 5]
```diff
@@ -1,545 +1,132 @@
...
 
-### 3.10 protocols/v2/noise-sv2
+### 🛠️Pools
```
Comment (GitGab19) [inline README.md:60]: Space in between

File: README.md [hunk 6]
```diff
@@ -1,545 +1,132 @@
...
 
-Contains the Noise Protocol logic.
-TODO: More information.
+- Pools supporting Sv2 can deploy the open source binary crate (`roles/pool`) to offer their clients (miners participating in said pool) an Sv2-compatible pool.
+- The Rust helper library provides a suite of tools for mining pools to build custom Sv2 compatible pool implementations.
+- The C library provides a set of FFI bindings to the Rust helper library for miners to integrate Sv2 into their existing firmware stack.
```
Comment (GitGab19) [inline README.md:64]: Uppercase SV2 (?)

File: README.md [hunk 7]
```diff
@@ -1,545 +1,132 @@
...
 
-**External dependencies**: TODO
-**Internal dependencies**: TODO
+## 🛣Roadmap
```
Comment (GitGab19) [inline README.md:66]: Space

File: README.md [hunk 8]
```diff
@@ -1,545 +1,132 @@
...
 
-- roles properties
-- message handlers
-- channel "routing" logic (Group and Extended channels)
-- `Sv2Frame` <-> specific (sub)protocol message mapping
-- (sub)protocol <-> (sub)protocol mapping
-- job logic (this overlaps with the channel logic)
-- Bitcoin data structures <-> Sv2 data structures mapping
-- utils
+### 🎯Goals
```
Comment (GitGab19) [inline README.md:72]: Space

File: README.md [hunk 9]
```diff
@@ -1,545 +1,132 @@
...
-To build the various projects/examples follow the following commands:
+1. A robust set of Stratum V2 (Sv2) primitives as Rust library crates which anyone can use
+   to expand the protocol or implement a role. For example:
+   - Pools supporting Sv2
+   - Mining-device/hashrate producers integrating Sv2 into their firmware
+   - Bitcoin nodes implementing Template Provider to build the `blocktemplate`
+2. The above Rust primitives as a C library available for use in other languages via FFI.
+3. A set of helpers built on top of the above primitives and the external Bitcoin-related Rust crates for anyone to implement the Sv2 roles.
+4. An open-source implementation of a Sv2 proxy for miners.
+5. An open-source implementation of a Sv2 pool for mining pool operators.
```
Comment (GitGab19) [inline README.md:84]: Uppercase SV2 (?)

File: README.md [hunk 10]
```diff
@@ -1,545 +1,132 @@
...
 
-`cargo build` - This quickly builds a test/debug artifacts for all crates
+## 💻Contribute
```
Comment (GitGab19) [inline README.md:86]: Space

File: README.md [hunk 11]
```diff
@@ -1,545 +1,132 @@
...
- | [async_std](https://async.rs/)                        | Alternative to tokio                                                                                | yes     | network_helpers                                                                                              |
- | with_[tokio](https://tokio.rs/)                       | For async with custom (de)serializer                                                                | no      | network_helpers                                                                                              |
- | debug                                                 | Turns on debugging features                                                                         | no      | utils/buffer                                                                                                 |
- | [criterion](https://github.com/bheisler/criterion.rs) | To [run benchmark tests](utils/buffer/README.md)                                                    | no      | utils/buffer                                                                                                 |
+## 🤝Support
```
Comment (GitGab19) [inline README.md:92]: Space

File: README.md [hunk 12]
```diff
@@ -1,545 +1,132 @@
...
 
-#### Building/Updating sv2-ffi
+Join our Discord community to get help, share your ideas, or discuss anything related to Stratum V2 and its refference implementation.
```
Comment (GitGab19) [inline README.md:94]: Typo: reference

File: README.md [hunk 13]
```diff
@@ -1,545 +1,132 @@
...
 
-If the sv2.h file is out of sync then the following steps can be taken to update it:
-From the root directory:
-`$ ./build_header.sh`
-This script will install cbindgen and then generate the `sv2.h` header file. This header file needs to be copied to the
-`protocols/v2/sv2-ffi` directory and committed.
+[Join the Stratum V2 Discord Community](https://discord.gg/fsEW23wFYs)
 
-### 4.02 Test
+## 🎁Donate
```
Comment (GitGab19) [inline README.md:100]: Space

File: README.md [hunk 14]
```diff
@@ -1,545 +1,132 @@
...
 
-As a prerequisite, you must have [`cargo-tarpaulin`](https://github.com/xd009642/tarpaulin) installed globally:
+### 👤Individual Donations
```
Comment (GitGab19) [inline README.md:102]: Space

File: README.md [hunk 15]
```diff
@@ -1,545 +1,132 @@
...
-```
-$ cargo install cargo-tarpaulin
-```
+[Donate through OpenSats](https://opensats.org/projects/stratumv2)
 
-Generate test coverage percentage with cargo-tarpaulin:
-```
-$ cargo +nightly tarpaulin --verbose
-```
+### 🏢Corporate Donations
```
Comment (GitGab19) [inline README.md:107]: Space

File: README.md [hunk 16]
```diff
@@ -1,545 +1,132 @@
...
-- `features = "..."`: Includes the code with the listed features. The following features result in a `tarpaulin` error and are NOT included: `derive`, `alloc`, `arbitrary-derive`, `attributes`, and `with_serde`
-- `run-types = [ "Lib" ]`: Only tests the package's library unit tests. Includes protocols, and utils (without the exclude-files flag, it includes this example because it contains a `lib.rs` file)
-- `exclude-files = [ "examples/*" ]`: Excludes all projects in examples directory (specifically added to ignore examples that that contain a `lib.rs` file like `interop-cpp`)
-- `timeout = "120s"`: If unresponsive for 120 seconds, action will fail
-- `fail-under = 20`: If code coverage is less than 20%, action will fail
-- `out = ["Xml"]`: Required for `codecov.io` to generate coverage result
+Email us at: stratumv2@gmail.com
 
-### 4.03 Run
+## 🙏Supporters
```
Comment (GitGab19) [inline README.md:112]: Space

File: README.md [hunk 17]
```diff
@@ -1,545 +1,132 @@
...
 
-To get a list of the available binaries that can be run type `cargo run --bin` from the root. Then to run specify the 
-binary - for example from the table below to run the interop-cpp test you'd run `cargo run --bin interop-cpp`
+SRI contributors are independently, financually supported by following entities:
```
Comment (GitGab19) [inline README.md:114]: Typo: financially

File: README.md [hunk 18]
```diff
@@ -1,545 +1,132 @@
...
-| test-pool               | roles/v2/test-utils/pool          | Used in the sv2-proxy example as the sv2 pool     |
- 
+<p float="left">
+  <a href="https://hrf.org"><img src="https://raw.githubusercontent.com/stratum-mining/stratumprotocol.org/main/src/.vuepress/public/assets/hrf-logo-boxed.svg" width="250" /></a>
+  <a href="https://spiral.xyz"><img src="https://raw.githubusercontent.com/stratum-mining/stratumprotocol.org/main/src/.vuepress/public/assets/Spiral-logo-boxed.svg" width="250" /></a>
+  <a href="https://opensats.org/"><img src="https://raw.githubusercontent.com/stratum-mining/stratumprotocol.org/main/src/.vuepress/public/assets/opensats-logo-boxed.svg" width="250" /></a>
+</p>
 
-## 5. Branches
+## 📖License
```
Comment (GitGab19) [inline README.md:122]: Space

**Review by GitGab19 — COMMENTED** [review]
I really like the new structure, it's very clear for externals!
I left some typos and corrections/suggestion

File: README.md (hunk 1 above)
Comment (pavlenex) [reply README.md]: Good catch, I've fixed this

File: README.md (hunk 4 above)
Comment (pavlenex) [reply README.md]: Good catch, done.

**Comment by pavlenex** [comment]
@GitGab19 Thanks for a review, I've added spacing with emojies, the only compromise there is that I had to add an ugly link for getting started guide, but that's a minor thing. I've also fixed few typos I found and the one you reported.

File: README.md (hunk 17 above)
Comment (pavlenex) [reply README.md]: Good catch, fixed

**Comment by pavlenex** [comment]
@Fi3 when you get a chance, review and merge.

**Pull request merged**